src/software/geom/util.cpp

```diff
@@ -507,7 +508,10 @@ std::vector<Segment> getEmptySpaceWithinParentSegment(std::vector<Segment> segme
     for (std::vector<Segment>::const_iterator it = open_segs.begin();
          it != open_segs.end();)
     {
-        if (it->length() < EPS)
+        // If segment length is less than 2 * FIXED_EPSILON, remove the segment
+        // 2 * FIXED_EPSILON chosen as FIXED_EPSILON by itself is not a sufficient for all
+        // cases
```

Comment:
Sounds like a tautological justification. Can we investigate this issue to see if there are reasons why we're accumulating so much error? For example, two floating point operations would mean we could expect up to 2 \* FIXED_EPSILON error. See test_util.cpp:equalWithinTolerance for some examples.

Reply:
Improved comment, let me know if there's any issues in the reasoning.